### PR TITLE
Update drake-ros CI workflow 

### LINKED ISSUE #26: ci: Not currently running on `develop`? Possible to see badges?
Current workflows on `develop` are not running.
https://github.com/RobotLocomotion/drake-ros/actions

Possible to see badges?

\cc @IanTheEngineer

### PR DESCRIPTION
**Pull request by hidmic**
Closes #26. Should be rebase-merged, not squashed-merged.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/28)

### CODE REVIEW

**Comment by hidmic** [comment]
+@EricCousineau-TRI for review

File: README.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
-# drake_ros
+# Drake ROS
+
+[![drake-ros continuous integration](https://github.com/RobotLocomotion/drake-ros/actions/workflows/main.yml/badge.svg?branch=develop)](https://github.com/RobotLocomotion/drake-ros/actions/workflows/main.yml)
```
Comment (hidmic) [inline README.md:3]: FYI it doesn't look like we can get a different badge shield per branch.

Comment (EricCousineau-TRI) [reply README.md:3]: Huh? I see two diff. badges for these two URLs:
```
https://github.com/RobotLocomotion/drake-ros/actions/workflows/main.yml/badge.svg?branch=develop
https://github.com/RobotLocomotion/drake-ros/actions/workflows/main.yml/badge.svg?branch=hidmic/ci-tweaks
```

Comment (EricCousineau-TRI) [reply README.md:3]: I also see a useful badge for your separate PR:
```
https://github.com/RobotLocomotion/drake-ros/actions/workflows/bazelized.yml/badge.svg?branch=hidmic/drake_ros_bazel_installed
```

**Review by EricCousineau-TRI — APPROVED** [review]
:lgtm_strong: 

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/28#-:-Mo0gS9B9dDJwRfKv88l:b-uwkowd)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @hidmic)

**Comment by EricCousineau-TRI** [comment]
(I'll leave it up to you or Ian to merge)

File: README.md (hunk 1 above)
Comment (hidmic) [reply README.md:3]: Right, but the badge itself is the same. The text is taken from the workflow name, regardless of which branch it was targeting.

Comment (EricCousineau-TRI) [reply README.md:3]: Oh, gotcha. Yeah, I think that's fine.
Ideally, whatever branch is shown, we just shown that status.

And yeah, if we want to show develop, I think it's fine to just make a table; perhaps use the `shields.io` variant for something a bit more compact.

Comment (hidmic) [reply README.md:3]: > perhaps use the shields.io variant for something a bit more compact.

[GitHub Workflow Status (branch)](https://github.com/RobotLocomotion/drake-ros/actions/workflows/main.yml)

[GitHub Workflow Status (branch)](https://github.com/RobotLocomotion/drake-ros/actions/workflows/main.yml)

Neat! Shall we switch to it?

Comment (EricCousineau-TRI) [reply README.md:3]: Sounds good to me!

Comment (hidmic) [reply README.md:3]: Ah, I just saw this. I'll do so in a follow-up PR.

**Pull request merged**